### PR TITLE
docs: update release instruction

### PR DESCRIPTION
**Pull request by roman-khimov**
It can be simplified after b27e6918bd969cc07f738ba96c7f246f44ede319.

### CODE REVIEW

**Review by AnnaShaleva — DISMISSED** [review]

File: docs/release-instruction.md [hunk 1]
```diff
@@ -23,31 +24,27 @@ optimizations, refactoring and API changes. Add a "Behaviour changes" section
...
+## Add automatically-built binaries
 
-## Make a proper Github release
-
-Edit an automatically-created release on Github, copy things from changelog
-there and attach previously downloaded binary. Make a release.
+New release created at the previous step triggers automatic builds (if not,
+start them manually from the Build Github workflow), so wait for them to
+finish. Then download currently supported binaries (at the time of writing
+that's `neo-go-macos-12-arm64`, `neo-go-ubuntu-20.04-amd64`,
```
Comment (AnnaShaleva) [inline docs/release-instruction.md:42]: Not sure whether it's important, but we have `amd64` for macOS also.

File: docs/release-instruction.md [hunk 2]
```diff
@@ -23,31 +24,27 @@ optimizations, refactoring and API changes. Add a "Behaviour changes" section
...
+`neo-go-ubuntu-20.04-arm64` and `neo-go-windows-2022-amd64`), unpack archives and
+rename binaries from them using `neo-go-$GOOS-$GOARCH` scheme, at the moment
+that'd be `neo-go-darwin-arm64`, `neo-go-linux-amd64`, `neo-go-linux-arm64`
+and `neo-go-windows-amd64.exe`. Add these binaries to the previously created
```
Comment (AnnaShaleva) [inline docs/release-instruction.md:46]: I've updated our building job at #2727. You don't need to rename binaries anymore. Also, archives names have been changed, although I'm not sure about it and may rollback this change.

Here's the example of resulting archives and binaries: https://github.com/nspcc-dev/neo-go/actions/runs/3194559297.

File: docs/release-instruction.md (hunk 1 above)
Comment (roman-khimov) [reply docs/release-instruction.md:42]: Yes, we specifically do not release it. No one has tested it and people are moving to `arm64` anyway.

File: docs/release-instruction.md (hunk 2 above)
Comment (roman-khimov) [reply docs/release-instruction.md:46]: Yeah, this can simplify things a bit.

Comment (roman-khimov) [reply docs/release-instruction.md]: Updated.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**